sparkles/core.py

```diff
@@ -700,19 +730,12 @@ def __init__(self, *args, **kwargs):
     def t_ccds_bonus(self):
         """Effective T_ccd for each guide star, including dynamic background bonus."""
         if not hasattr(self, "_t_ccds_bonus"):
-            # Compute guide count once for the record
-            mags = self.guides["mag"]
-            t_ccds = np.full_like(mags, self.guides.t_ccd)
-            if self.dyn_bgd_n_faint > 0:
-                # Apply the dynamic background t_ccd bonus to the
-                # dyn_bgd_n_faint faintest stars, ensuring that at least
-                # MIN_DYN_BGD_ANCHOR_STARS are evaluated without the bonus. See:
-                # https://nbviewer.org/urls/cxc.harvard.edu/mta/ASPECT/ipynb/misc/guide-count-dyn-bgd.ipynb
-                mags = np.sort(mags)
-                n_faint = min(self.dyn_bgd_n_faint, len(t_ccds))
-                idx_bonus = max(len(t_ccds) - n_faint, MIN_DYN_BGD_ANCHOR_STARS)
-                t_ccds[idx_bonus:] += self.dyn_bgd_dt_ccd
-            self._t_ccds_bonus = t_ccds
+            self._t_ccds_bonus = get_t_ccds_bonus(
```

Comment:
My only concern is that self.mags is not guaranteed to be in the same order as self.guides['mag'].  This isn't a logical issue but suggests a bit to me that the t_ccds_bonus should be an attribute of self.guides not self.

Reply:
Given urgency and that having the bonus "top level" is not a change in this PR, this is just an aside for future consideration.

Reply:
Regarding "That's not a thing in this code as far as I can see." I agree.  But note that there is a "self.mags" that is the mags of the full aca review table that is distinct from self.guides['mags'].

Reply:
You probably mean `self["mag"]`?

Reply:
By "having the bonus 'top level'" I just meant that the t_ccds_bonus property is on self and not self.guides and that is not a change introduced in this PR.  I think we're on the same page here.

Reply:
And "You probably mean self["mag"]?" and yes, I was being imprecise.  I did mean self['mag'] not self.mags.